datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -785,180 +786,166 @@ mod tests {
     use crate::physical_plan::udf::ScalarUDF;
 
     #[test]
-    fn test_simplify_or_true() -> Result<()> {
-        let expr_a = col("c").or(lit(true));
-        let expr_b = lit(true).or(col("c"));
+    fn test_simplify_or_true() {
+        let expr_a = col("c2").or(lit(true));
+        let expr_b = lit(true).or(col("c2"));
         let expected = lit(true);
 
         assert_eq!(simplify(&expr_a), expected);
         assert_eq!(simplify(&expr_b), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_or_false() -> Result<()> {
-        let expr_a = lit(false).or(col("c"));
-        let expr_b = col("c").or(lit(false));
-        let expected = col("c");
+    fn test_simplify_or_false() {
+        let expr_a = lit(false).or(col("c2"));
+        let expr_b = col("c2").or(lit(false));
+        let expected = col("c2");
 
         assert_eq!(simplify(&expr_a), expected);
         assert_eq!(simplify(&expr_b), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_or_same() -> Result<()> {
-        let expr = col("c").or(col("c"));
-        let expected = col("c");
+    fn test_simplify_or_same() {
+        let expr = col("c2").or(col("c2"));
+        let expected = col("c2");
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_and_false() -> Result<()> {
-        let expr_a = lit(false).and(col("c"));
-        let expr_b = col("c").and(lit(false));
+    fn test_simplify_and_false() {
+        let expr_a = lit(false).and(col("c2"));
+        let expr_b = col("c2").and(lit(false));
         let expected = lit(false);
 
         assert_eq!(simplify(&expr_a), expected);
         assert_eq!(simplify(&expr_b), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_and_same() -> Result<()> {
-        let expr = col("c").and(col("c"));
-        let expected = col("c");
+    fn test_simplify_and_same() {
+        let expr = col("c2").and(col("c2"));
+        let expected = col("c2");
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_and_true() -> Result<()> {
-        let expr_a = lit(true).and(col("c"));
-        let expr_b = col("c").and(lit(true));
-        let expected = col("c");
+    fn test_simplify_and_true() {
+        let expr_a = lit(true).and(col("c2"));
+        let expr_b = col("c2").and(lit(true));
+        let expected = col("c2");
 
         assert_eq!(simplify(&expr_a), expected);
         assert_eq!(simplify(&expr_b), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_multiply_by_one() -> Result<()> {
-        let expr_a = binary_expr(col("c"), Operator::Multiply, lit(1));
-        let expr_b = binary_expr(lit(1), Operator::Multiply, col("c"));
-        let expected = col("c");
+    fn test_simplify_multiply_by_one() {
```

Comment:
Maybe you can also add `test_simlify_multiply_by_zero` -> 0 ？

Reply:
I tried that, and it turns out that rewrite rule is not implemented LOL 👍 

```diff
diff --git a/datafusion/src/optimizer/simplify_expressions.rs b/datafusion/src/optimizer/simplify_expressions.rs
index 3c4af838b..68d45c446 100644
--- a/datafusion/src/optimizer/simplify_expressions.rs
+++ b/datafusion/src/optimizer/simplify_expressions.rs
@@ -851,6 +851,16 @@ mod tests {
         assert_eq!(simplify(&expr_b), expected);
     }
 
+    #[test]
+    fn test_simplify_multiply_by_zero() {
+        let expr_a = binary_expr(col("c2"), Operator::Multiply, lit(0));
+        let expr_b = binary_expr(lit(0), Operator::Multiply, col("c2"));
+        let expected = lit(0);
+
+        assert_eq!(simplify(&expr_a), expected);
+        assert_eq!(simplify(&expr_b), expected);
+    }
+
     #[test]
     fn test_simplify_divide_by_one() {
         let expr = binary_expr(col("c2"), Operator::Divide, lit(1));
```

Resulted in 

```
---- optimizer::simplify_expressions::tests::test_simplify_multiply_by_zero stdout ----
thread 'optimizer::simplify_expressions::tests::test_simplify_multiply_by_zero' panicked at 'assertion failed: `(left == right)`
  left: `#c2 * Int32(0)`,
 right: `Int32(0)`', datafusion/src/optimizer/simplify_expressions.rs:860:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Tracking in https://github.com/apache/arrow-datafusion/issues/1406